tests/Tokenizer/TokenTest.php

```diff
@@ -346,6 +346,12 @@ public function provideEqualsCases(): array
             // if it is an array any additional field is checked too
             [$function, false, [T_FUNCTION, 'function', 'unexpected']],
         ];
+
+        yield [new Token('&'), true, '&'];
+        if (\defined('T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG')) { // @TODO: drop condition with new MAJOR release 4.0
+            yield [new Token([T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG, '&']), true, '&'];
+            yield [new Token([T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG, '&']), true, '&'];
+        }
```

Comment:
please add test for combinations:
- `&` equals `T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG`
- `&` equals `T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG`
- `T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG` equals `&`
- `T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG` equals `T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG`
- `T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG` equals `&`
- `T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG` equals `T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG`

Reply:
added